app/main.py

```diff
@@ -1,2 +1,58 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hash_table = self.make_hash_table()
+
+    def make_hash_table(self) -> list:
+        return [[] for _ in range(self.capacity)]
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_key = hash(key)
+        hash_index = hash_key % self.capacity
+        while True:
+            try:
+                if (
+                        self.hash_table[hash_index][0] == key
+                        and self.hash_table[hash_index][1] == hash_key
+                ):
+                    return self.hash_table[hash_index][2]
+            except IndexError:
+                raise KeyError
+            hash_index = (hash_index + 1) % self.capacity
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.size = 0
+        self.capacity *= 2
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hash_table = self.make_hash_table()
+        for bucket in old_hash_table:
+            if bucket:
+                self.__setitem__(bucket[0], bucket[2])
```

Comment:
There's a certain shortcut for the `__setitem__()` method you are definitely aware of. You might as well use it.

Reply:
I don't understand so much about how to make my code better, I think it works pretty well enough, maybe after the mentor review I'll have to change smth, but for now, I think it's ok) thx for the review

Reply:
What I'm talking about in this comment is `self.[bucket[0]] = bucket[2]`. Since it's a magic method, you have no real reason to call it directly, there's a shortcut.